### PR TITLE
Update CHANGELOG for 1.6.0

### PR DESCRIPTION
**Pull request by utpilla**
## Changes
- Update CHANGELOG for `1.6.0`
- Update public API files
- Updated CHANGELOG for `OpenTelemetry.Shims.OpenTracing` on auto-instrumentation team's request. They asked for a newer release for this package

### CODE REVIEW

**Review by reyang — APPROVED** [review]

File: src/OpenTelemetry/CHANGELOG.md [hunk 1]
```diff
@@ -2,12 +2,24 @@
...
+Released 2023-Sep-05
+
 * Increased the character limit of the Meter instrument name from 63 to 255.
   ([#4774](https://github.com/open-telemetry/opentelemetry-dotnet/issues/4774))
 
 * Update default size for `SimpleExemplarReservoir` to `1`.
   ([#4803](https://github.com/open-telemetry/opentelemetry-dotnet/issues/4803))
 
+* Update Metrics SDK to override the default histogram bounds for a set of
+  well-known histogram metrics from AspNetCore and Http runtime and their
```
Comment (vishweshbankwar) [inline src/OpenTelemetry/CHANGELOG.md:16]:
```suggestion
  well-known histogram metrics for ASP.NET Core and HttpClient
```

Comment (vishweshbankwar) [reply src/OpenTelemetry/CHANGELOG.md:16]: I think it is ok to skip the instrumentation library part (We will add details on instrumentation libraries side on how to enable these bounds).

File: src/OpenTelemetry/CHANGELOG.md [hunk 2]
```diff
@@ -2,12 +2,24 @@
...
+  corresponding instrumentation libraries. These histogram metrics which have
+  their `Unit` as `s` (second) will have their default histogram buckets as `[
+  0, 0.005, 0.01, 0.025, 0.05, 0.075, 0.1, 0.25, 0.5, 0.75, 1, 2.5, 5, 7.5, 10
+  ]`.
+  ([#4820](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4820))
```
Comment (vishweshbankwar) [inline src/OpenTelemetry/CHANGELOG.md:21]: nit: Could update the PR separately to include the list of metric names in description.

**Review by vishweshbankwar — APPROVED** [review]

**Review by rajkumar-rangaraj — APPROVED** [review]

File: src/OpenTelemetry/CHANGELOG.md (hunk 2 above)
Comment (utpilla) [reply src/OpenTelemetry/CHANGELOG.md:21]: Yes. I commented about the same in the original PR: https://github.com/open-telemetry/opentelemetry-dotnet/pull/4820#discussion_r1316475276

**Pull request merged**